### ISSUE TITLE
LocalVolumeType: gce-localssd-scsi-fs tests are failing

### ISSUE THREAD

**Issue by msau42** [kubernetes/kubernetes#74384]
**Which jobs are failing**:
ci-kubernetes-e2e-gci-gce-serial

**Which test(s) are failing**:
LocalVolumeType: gce-localssd-scsi-fs

**Since when has it been failing**:
2/20

**Testgrid link**:
https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/4710

**Reason for failure**:
unsure yet

/assign

**Comment by msau42** [comment]
@kubernetes/sig-storage-test-failures

**Comment by msau42** [comment]
Looks like reconstruction is not correctly implemented for local PVs:
```
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: W0223 01:29:59.091766    4465 reconciler.go:394] Could not construct volume information, cleanup the mounts. (pod.UID 6e211672-370a-11e9-b9d7-42010a
800002, volume.SpecName local-4ws2k): reconstructVolume.NewMounter failed for volume "kubernetes.io/local-volume/local-4ws2k" (spec.Name: "local-4ws2k") pod "6e211672-370a-11e9-b9d7-42010a800002" (UID: "6e211672
-370a-11e9-b9d7-42010a800002") with: local volume source is nil or local path is not set
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: I0223 01:29:59.091775    4465 reconciler.go:429] Reconciler sync states: could not find information (PID: 6e211672-370a-11e9-b9d7-42010a800002) (Vol
ume SpecName: local-4ws2k) in desired state, clean up the mount points
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: E0223 01:29:59.091883    4465 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory ad
dress or nil pointer dereference)
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /usr/local/go/src/runtime/asm_amd64.s:522
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /usr/local/go/src/runtime/panic.go:513
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /usr/local/go/src/runtime/panic.go:82
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /usr/local/go/src/runtime/signal_unix.go:390
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/volume/util/operationexecutor/operation_generator.go:1538
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/volume/util/operationexecutor/operation_generator.go:735
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/volume/util/operationexecutor/operation_executor.go:774
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler/reconciler.go:440
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler/reconciler.go:395
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler/reconciler.go:330
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler/reconciler.go:155
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
Feb 23 01:29:59 e2e-test-msau-minion-group-qfrw kubelet[4465]: /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
```

1. Need to check if it's ok for mounter to be initialized with an empty path
2. Investigate the panic in the operation_generator

**Comment by msau42** [comment]
cc @cofyc 

Unsure why it's failing only for this local pv type (real disk vs loopback block)

I have not investigated the 2nd test yet

**Comment by spiffxp** [comment]
/priority critical-urgent
/milestone v1.14
We're entering [burndown](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.14#burndown) and this looks relevant to the 1.14 release.  Please `/milestone clear` if I am incorrect

**Comment by cofyc** [comment]
hi, checked [latest test results](https://testgrid.k8s.io/sig-storage-kubernetes#gce-serial&include-filter-by-regex=Driver%3A%20local&include-filter-by-regex=Driver%3A%20local), after #74445, "should be mountable" has been fixed.

failing test is "should unmount if pod is force deleted while kubelet is down [Disruptive][Slow]" and it fails for all kinds of local volume types

**Comment by cofyc** [comment]
> 2\. Investigate the panic in the operation_generator

This is because local volume cannot be reconstructed for now (localPlugin.NewMounter requires `Path` field is set), so `reconciler.syncStates` is going to [cleanup mounts for it](https://github.com/kubernetes/kubernetes/blob/139a13d312b1b11de7f3af63d9503365ff3a1e6a/pkg/kubelet/volumemanager/reconciler/reconciler.go#L393-L396). But `operationGenerator.GenerateUnmountVolumeFunc` uses `useCSIPlugin` to check `volumeToUnmount.VolumeSpec` if it should use CSI plugin, but `volumeToUnmount.VolumeSpec` is not set in this case, so it panics here.

I'm not sure how to support reconstruction for local volumes correctly. But we can fix this panic first, then reconciler will clean up the existing mount points if the volume is no longer needed. IIUC, this will fix this failing test.

https://github.com/kubernetes/kubernetes/blob/139a13d312b1b11de7f3af63d9503365ff3a1e6a/pkg/kubelet/volumemanager/reconciler/reconciler.go#L363-L368

**Comment by mariantalla** [comment]
I see similar failures in [sig-release-master-upgrade#gce-new-master-upgrade-cluster-new](https://testgrid.k8s.io/sig-release-master-upgrade#gce-new-master-upgrade-cluster-new) and [sig-release-1.14-blocking#1.14-blocking](https://testgrid.k8s.io/sig-release-1.14-blocking#gce-cos-1.14-serial) too; are they connected?

cc @alejandrox1 @smourapina @kacole2 @mortent

**Comment by msau42** [comment]
yes they should be similar

**Comment by alejandrox1** [comment]
need to keep track of this to see if failures in [master-blocking#gce-cos-master-serial](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-serial) also clear up.

**Comment by mariantalla** [comment]
The tests are still throwing `panic`s in [1.14-blocking](https://testgrid.k8s.io/sig-release-1.14-blocking#gce-cos-1.14-serial) (it's the `ci-kubernetes-e2e-gce-cos-k8sbeta-serial` job there though).

They also seem to not have run since the 24th, I'll look into that in the meantime 🤔
